api/tests/mocha/controllers/login.spec.js

```diff
@@ -109,9 +110,11 @@ describe('login controller', () => {
       const getUserCtx = sinon.stub(auth, 'getUserCtx').resolves({ name: 'josh' });
       const redirect = sinon.stub(res, 'redirect');
       const cookie = sinon.stub(res, 'cookie').returns(res);
+      const getDoc = sinon.stub(db, 'get').resolves({ doc: { _id: 'branding' }});
       controller.get(req, res).then(() => {
         chai.expect(getUserCtx.callCount).to.equal(1);
         chai.expect(getUserCtx.args[0][0]).to.deep.equal(req);
+        chai.expect(getDoc.callCount).to.equal(1);
```

Comment:
I don't think this is correct. From the code it looks like `getDoc` should only be called if `getUserCtx` throws. Why's it being called here?

Reply:
I think the problem may be that this test doesn't return a promise or call a callback function, so I think the test is passing no matter what happens inside the `controller.get(req, res).then(() => {` block. You can test this out by changing any of the expected values and making sure the test fails.

I think you need to return the promise returned from the `controller.get` function, eg:

```js
return controller.get(req, res).then(() => {
```

This also applies to the test below so check all the tests in this file and make sure they're actually working!

Reply:
@garethbowen please review